hedera-node/src/main/java/com/hedera/services/context/SideEffectsTracker.java

```diff
@@ -274,6 +273,24 @@ public void trackAutoAssociation(final TokenID token, final AccountID account) {
 		autoAssociations.add(association);
 	}
 
+	/**
+	 * Tracks the reward payment to the account.
+	 *
+	 * @param accountNum
+	 * 		Account number of the account receiving the reward.
+	 * @param amount
+	 * 		Reward amount.
+	 */
+	public void trackRewardPayment(final long accountNum, final long amount) {
+		if (rewardAmounts.length == 0 || rewardedAccounts.length == 0) {
+			rewardAmounts = new long[BALANCE_CHANGES_LENGTH];
+			rewardedAccounts = new long[BALANCE_CHANGES_LENGTH];
+		}
+		rewardedAccounts[numRewardedAccounts] = accountNum;
```

Comment:
This might duplicate `accountNum` in the `rewardedAccounts` list. We can easily avoid this by replacing lines 289-291 with,
```
numRewardedAccounts = includeOrderedFungibleChange(
    rewardedAccounts, rewardAmounts, numRewardedAccounts, accountNum, amount);
```